dbt-snowflake/src/dbt/adapters/snowflake/parse_model.py

```diff
@@ -0,0 +1,98 @@
+from typing import Iterable, Optional
+
+from dbt_common.exceptions import DbtConfigError
+from dbt.adapters.contracts.relation import RelationConfig
+
+from dbt.adapters.snowflake import constants
+
+
+def automatic_clustering(model: RelationConfig) -> Optional[bool]:
+    if not model.config:
+        return None
+
+    return model.config.get("automatic_clustering", False)
```

Comment:
```suggestion
    if model.config:
        return model.config.get("automatic_clustering", False)
    else:
        return None
```

Reply:
The initial check is really a `mypy` thing. Because `config` is optional, `mypy` raises an error when trying to access it. I started on more complicated methods where I didn't want a bunch of logic inside of an if block; I then copy/pasted that structure to the rest of the methods. So I could see an argument for consistency, but this also works.